library/src/main/java/com/google/android/exoplayer/hls/Variant.java

```diff
@@ -27,10 +27,15 @@
   public final String url;
   public final Format format;
 
-  public Variant(int index, String url, int bitrate, String codecs, int width, int height) {
+  public Variant(int index, String name, String url, int bitrate, String codecs, int width, int height) {
     this.url = url;
-    format = new Format(Integer.toString(index), MimeTypes.APPLICATION_M3U8, width, height, -1, -1,
-        -1, bitrate, null, codecs);
+    if (name != null) {
```

Comment:
Meh. I don't really want to start adding extra fields, unless there's some kind of dire need for them. In the worst case we might end up four ids (index-in-manifest, index-in-container, name-in-manifest, name-in-container), and it all starts getting a bit out of hand :). Let's just leave it how you have it for now and see if works out ok. Although as a nit, can we change this to be a bit more concise, like:

String formatName = name != null ? name : Integer.toString(index);
format = new Format(formatName, MimeTypes.APPLICATION_M3U8, width, height, -1, -1,
    -1, bitrate, null, codecs);

Thanks!